runtime/appruntime/metrics/json_based/json_based.go

```diff
@@ -0,0 +1,141 @@
+package json_based
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog"
+
+	"encore.dev/appruntime/config"
+	"encore.dev/metrics"
+)
+
+func New(svcs []string, cfg *config.JSONBasedMetricsProvider, rootLogger zerolog.Logger) *Exporter {
```

Comment:
I think the JSON metrics should not be thought of as a metrics provider/exporter, but as simply a pure function that takes `[]metrics.CollectedMetric` and returns a data a data structure. Essentially what you have with `(*Exporter).GetMetricData`. But we don't need to treat it as a "JSONBasedMetricsProvider" or change the configuration, I don't think. That way exposing metrics via API is orthogonal to the notion of a metrics exporter. (Thanks @campesel for pointing this out!)